src/main/java/org/craftercms/studio/controller/rest/v2/ClusterManagementController.java

```diff
@@ -60,6 +57,16 @@ public ResponseBody removeClusterMembers(@RequestParam(value = "id") List<Long>
         return responseBody;
     }
 
+    @PostMapping("/api/2/cluster/setPrimary")
+    public ResponseBody setClusterPrimary() {
+        boolean success = clusterManagementService.setClusterPrimary();
```

Comment:
Surround with try/catch/finally and perhaps log?

Reply:
Added try/catch/finally.